gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageStatistics.java

```diff
@@ -23,27 +23,86 @@
 import com.google.common.collect.Maps;
 import java.util.EnumSet;
 
-/** Statistics which are collected in GCS Connector */
+/** Statistics which are collected in GCS Connector. */
 public enum GoogleCloudStorageStatistics {
-
-  /** GCS connector specific statistics */
-  GCS_REQUEST_COUNT(
-      "gcs_total_request_count", "Counts the total number of gcs requests made", TYPE_COUNTER),
-
   EXCEPTION_COUNT("exception_count", "Counts the number of exceptions encountered", TYPE_COUNTER),
 
-  GCS_CLIENT_SIDE_ERROR_COUNT(
-      "gcs_client_side_error_count",
+  /** Status Code Counters for JSON Path */
+  GCS_API_REQUEST_COUNT(
+      "gcs_api_total_request_count", "Counts the total number of gcs requests made", TYPE_COUNTER),
+
+  GCS_API_CLIENT_SIDE_ERROR_COUNT(
+      "gcs_api_client_side_error_count",
       "Counts the occurrence of client side error status code",
       TYPE_COUNTER),
 
-  GCS_SERVER_SIDE_ERROR_COUNT(
-      "gcs_server_side_error_count",
+  GCS_API_SERVER_SIDE_ERROR_COUNT(
+      "gcs_api_server_side_error_count",
       "Counts the occurrence of server side error status code",
       TYPE_COUNTER),
 
-  GCS_CLIENT_RATE_LIMIT_COUNT(
-      "gcs_client_rate_limit_error_count", "Counts the occurence of 429 status code", TYPE_COUNTER);
+  GCS_API_CLIENT_RATE_LIMIT_COUNT(
+      "gcs_api_client_rate_limit_error_count",
+      "Counts the occurence of 429 status code",
```

Comment:
Is 429 the way gRPC reports ratelimiting. Maybe we can keep this a bit more generic so that it will apply for gRPC as well. Same for other metrics.